expose the number of attempts to the CAPA template

pdpinch opened the pull request.

pdpinch commented:
@jolyonb can you remind me how to test this manually?

The python tests passed at edX, but it failed a bunch of JavaScript tests. I'm not sure why. 

https://build.testeng.edx.org/job/edx-platform-js-pr/33625/testReport/
https://build.testeng.edx.org/job/edx-platform-bok-choy-pr/35278/testReport/

jolyonb commented:
Hi Peter,

Here's a codeblock that tests this:

```
<problem display_name="Submission Numbers">
<script type="text/python" system_path="python_lib">
def gradeit(expect, ans, **kwargs):
    stuff=str(kwargs)
    return {
        'input_list': [
            { 'ok': True, 'msg': stuff},
        ]
    }
</script>
<p>Any answer is correct.</p>
<customresponse cfn="gradeit" expect="1" options="test" cfn_extra_args="attempt">
  <textline size="45" correct_answer="1"/>
</customresponse>
</problem>
```

I have no idea why js stuff is failing. That makes little sense.

mahyard committed:
Prepend unicode signature (BOM) to csv reports
for MS Excel 2013 compatibility

awaisdar001 committed:
updated correct map when new score is calculated

Commit:
V1 Grades API with added support for bulk read of all user grades in a course.
URL patterns changed to follow edx API conventions

clemente committed:
Implement iter_all_for_block and iter_all_for_course for DjangoUserStateClient

Commit:
OC-3995
Moved the ID string used in the google-site-verification meta tag to settings so that it can be overridden from site_configuration.

hasnain-naveed committed:
WL-1518 | Get decline notification context from enterprise.

Mjloturco committed:
Add entitlement unenrollment survey

Updates behavior post unenrollment, also refactors accessible_modal
to enable the unenrollment survey to remain accessible after the
content in the modal changes (to the survey).

mloturco/learner-3524

brittneyexline committed:
ENT-945 Update IDVerification interfaces to accountfor SSOVerification

brittneyexline committed:
ENT-944 Create SSOVerifications for users in tpa pipeline based on provider's settings

jaebradley committed:
add cookie policy banner with feature flag

bmedx committed:
Add a model and associated API to help control the user retirement flow

bmedx committed:
Update retirement APIs to match spec

hamzamunir7300 committed:
Changing the contact-us link

Redirection to the newly created support form. If the server is production
it will take to production support form otherwise it will be forwarded to
Stage server support form.

LEARNER-4873

Commit:
LEARNER-4880 added confirmation modal for account deletion

Commit:
Add Notes Delete User API endpoint

Commit:
Updated the messages when a learner leave an organization.

Commit:
EDUCATOR-2628: deactivate and logout user

iloveagent57 committed:
Remove unused imports in user_api.models

saleem-latif committed:
Deleted SAML Identity provider appears in the Enterprise customer IdPs list.

Commit:
Revert "Show Enterprise display name if we should hide sensitive username."

This reverts commit d58bd06928c3c234e94cab24cb1e7971906067c1.

This commit caused performance issues due to increased API load from checking the
enterprise association on each page load

nasthagiri committed:
Remove Completion's usage of Course Structures API

Commit:
Add Enterprise middleware that injects customer data.

adeelkhan committed:
Catch HttpError thrown by Python Social Auth.

This patch enables catching HttpError exception raised
by PSA because of any issue caused at client/server.
Further it redirects user to login page with a
message about the error.

LEARNER-4344

Commit:
Merge pull request #17972 from edx/ai/EDUCATOR-2692-discussion

Discussion moderators, TA's ,admins see the posts according to cohorts.

jmbowman committed:
TE-2527 Move test and dev dependencies out of base.in

feanil committed:
Merge pull request #17967 from edx/gbabey/revert-ENT-924

Revert "Show Enterprise display name if we should hide sensitive user…

Commit:
Remove failing js test

Commit:
Merge pull request #17980 from edx/youngstrom/remove-failing-js-test

Remove failing js test on ff59

feanil committed:
Merge pull request #17981 from edx/release-mergeback-to-master

Merge release back to master

jaebradley committed:
Merge pull request #17941 from edx/LEARNER-4864-cookie-policy-banner-component

LEARNER-4864: Add Cookie Policy Banner

nasthagiri committed:
Remove Course Structure API

uzairr committed:
Merge pull request #17912 from edx/LEARNER-1487/send_successfull_verification_email

Send verification email status

ziafazal committed:
Merge pull request #17815 from edx/hasnain-naveed/WL-1518

WL-1518 | Get decline notification context from enterprise.

nasthagiri committed:
Merge pull request #17968 from edx/arch/remove-course-structure-usage-in-completion

Remove Completion's usage of Course Structures API

Commit:
Merge pull request #17975 from edx/ai/EDUCATOR-2719-button-styling

Fix styling of button on certificate & membership page.

waheedahmed committed:
Update support form to make course required.

Update support form to auto select course and make course
selection required to submit form.

LEARNER-4254

Rabia23 committed:
Merge pull request #17490 from edx/aj/updated-and-rescored-problem-still-shows-red-X

updated correct map when new score is calculated

Commit:
Changed the author info.

Commit:
Add passthrough for context_title and context_label LTI parameters

Commit:
Revert "pass title/label to LTI consumers"

This reverts commit f1fa7b8e06a3838c068549617b4df31f86e1131e.

Commit:
Updates to use the is_anonymous flag of basket calculate

[LEARNER-4913]

Commit:
Adds DeletableByUserValue to CourseEnrollmentAllowed model.

Commit:
Adds retire_user mixin.

Commit:
Merge pull request #17960 from edx/sstudent/educator-2628

EDUCATOR-2628: deactivate and logout user

jaebradley committed:
update cookie policy banner

Mjloturco committed:
retrap focus for seat unenroll survey a11y

learner-4715

Commit:
Merge pull request #17977 from edx/jmbowman/TE-2527

TE-2527 Move test and dev dependencies out of base.in

jaebradley committed:
Merge pull request #17989 from edx/update-cookie-policy-banner

Fix Cookie Policy Banner React Renderer Error

Commit:
Merge pull request #17965 from edx/efischer/lti_take2

pass context_title and _label to LTI provider

jmbowman committed:
TE-2547 Add dependency analysis scripts

Commit:
bump studio-frontend version to 1.7.3

saleem-latif committed:
Merge pull request #17956 from edx/saleem-latif/ENT-959

ENT-959: Deleted SAML Identity provider appears in the Enterprise customer IdPs list.

Commit:
Unlink SoftwareSecure PII for GDPR

Commit:
Use s3 db cache on devstack

amangano-edx committed:
Add password_policy checks to login views

LEARNER-4869

Mjloturco committed:
Merge pull request #17897 from edx/mloturco/learner-3524-accessible-modal

Add entitlement unenrollment survey, update accessible-modal

Commit:
Merge pull request #17994 from edx/fsheets/sfe-bump

Bump studio-frontend version to 1.7.3

Commit:
Merge pull request #17990 from edx/retireuserbyemailMixin

Adds DeletableByUserValue mixin.

Commit:
Merge pull request #17993 from edx/jmbowman/TE-2547

TE-2547 Add dependency analysis scripts

bmedx committed:
Clean up some warnings to get enough headroom to pass tests

iloveagent57 committed:
EDUCATOR-2693 | Delete PendingEmailChanges for retired users.

nedbat committed:
Merge pull request #14128 from edx/jbarciauskas/PLAT-1109-ignore-integrity-error

[PLAT-1109] Suppress IntegrityErrors caused by race condition

Commit:
Report the RELEASE_LINE in a meta tag

uzairr committed:
Send email for the failed verification status

Support Team need to handle bulk of tickets every month
about the verification status of learners.To avoid it,
suitable changes have done to send verification email
to the learner.

LEARNER-1488

ibrahimahmed443 committed:
Merge pull request #17936 from edx/ibrahimahmed443/WL-1519-duplicate-course-fix

catch duplicate course error

zubair-arbi committed:
ENT-960 Fix catalogs creation bug due to request data format

macdiesel committed:
Merge pull request #17698 from open-craft/clemente/implement-djangouserstateclient-iter-all

Implement iter_all_for_block and iter_all_for_course for DjangoUserStateClient

michaelroytman committed:
prevent loading both production and development CSS from the studio-frontend container

Commit:
Merge pull request #17997 from edx/youngstrom/enable-s3-cache-devstack

Use s3 db cache on devstack

bmedx committed:
Merge pull request #17919 from edx/bmedx/retirement_status

Add a model and associated API to help control the user retirement flow

michaelroytman committed:
add image accessibility best practice documentation URL and pass through to studio-frontend edit image modal

edx-cache-uploader-bot committed:
Updating migrations

uzairr committed:
Merge pull request #18002 from edx/LEARNER-1488/send_email_on_failed_verification

Send email for the failed verification status

Commit:
LEARNER-4700: Converts credentials to plugin

bmedx committed:
Add default retirement states to settings

Rabia23 committed:
update edx-proctoring version from 1.3.9 to 1.4.0.

michaelroytman committed:
Merge pull request #18004 from edx/mroytman/in-context-image-selection-css-fix

prevent loading both production and development CSS from the studio-f…

Commit:
Merge pull request #17988 from edx/unlink-courseenrollment-allowed

EDUCATOR-2688: Retire users in CourseEnrollmentAllowed model.

estute committed:
Merge pull request #18007 from edx/testeng/bokchoy_auto_cache_update_4117d52afc17c2ce0b091f8f0e252f42b1904161

Bokchoy db cache update

Rabia23 committed:
Merge pull request #18008 from edx/ri/edx-proctoring-version-1.4.0

update edx-proctoring version from 1.3.9 to 1.4.0.

Commit:
make userorgtag deletable

mduboseedx committed:
Merge pull request #17986 from open-craft/afzaledx/course_highlights_discovery

Changed the author info.

michaelroytman committed:
Merge pull request #18006 from edx/mroytman/add-accessibility-image-best-practice-doc-link

add image accessibility best practice documentation URL and pass thro…

Commit:
Creates oauth2_retirement_utils file, used it to retire oauth2 data.

bmedx committed:
Add a management command and settings to populate RetirementState models

UmanShahzad committed:
Merge pull request #17969 from edx/uman/enterprise-middleware

[ENT-962] Add Enterprise middleware that injects customer data.

nasthagiri committed:
Merge pull request #17983 from edx/arch/remove-course-structure-api

Remove Course Structure API

Commit:
Update translations

zubair-arbi committed:
Merge pull request #18003 from edx/zub/ENT-960-fix-catalogs-creation-bug

ENT-960 Fix catalogs creation bug due to request data format

noraiz-anwar committed:
fix outline progress

UmanShahzad committed:
Revert "Revert "Show Enterprise display name if we should hide sensitive user…"

Commit:
Add simulate_publish management command

Prior to this commit, any apps that update their data when courses are
published had to create their own management commands for bootstrapping
or error recovery (e.g. generate_course_overviews,
generate_course_blocks). This is a management command to allow us to
generically simulate a course_publish signal so that any async tasks
that trigger actions off of that can do so without having to each write
their own management commands.

It has a few options to make it more ops friendly:

* Can specify a set of courses, but defaults to all courses in the
  modulestore.
* Can specify a set of listeners, so we can bootstrap a new app without
  rebuilding everything.
* Can specify a delay between emitting signals so that we don't flood
  the queues and block author-initiated publishes from going through in
  a timely manner.
* Dry-run mode for a simple preview of what the script will attempt.

brittneyexline committed:
Merge pull request #17908 from edx/bexline/sso_id_verification

ENT-944 Create SSOVerifications for users in tpa pipeline based on provider's settings

irfanuddinahmad committed:
added admin site settings

edx-cache-uploader-bot committed:
Updating migrations

irfanuddinahmad committed:
Merge pull request #18021 from edx/iahmad/migrate_video_transcripts

added admin site settings

noraiz-anwar committed:
Merge pull request #18017 from edx/noraiz/EDUCATOR-2759

fix outline progress

Commit:
rebalance python unittests onto new shards

Commit:
Merge pull request #18022 from edx/testeng/bokchoy_auto_cache_update_d9252d990137c43d4aba1ea98d513f728cc1b735

Bokchoy db cache update

jmbowman committed:
TE-2546 Remove obsolete requirements files

iloveagent57 committed:
EDUCATOR-2632 | Management command for creating new DOT application model.

mulby committed:
support cross domain ajax requests to the experiments APIs

ssemenova committed:
Unlink CreditRequirementStatus PII for retired users

nedbat committed:
Merge pull request #18001 from edx/nedbat/release-line-meta-tag

Report the RELEASE_LINE in a meta tag

UmanShahzad committed:
Merge pull request #18018 from edx/revert-17967-gbabey/revert-ENT-924

[ENT-924] Revert "Revert "Show Enterprise display name if we should hide sensit…

hamzamunir7300 committed:
Merge pull request #17922 from edx/LEARNER-4873/fix-contact-us-link

Changing the contact-us link

Commit:
Merge pull request #17935 from edx/asadiqbal08/ENT-645

[ENT-645]- Updated the messages when a learner leave an organization.

Commit:
version bump

Commit:
Merge pull request #18031 from edx/asadiqbal08/version-bump

asadiqbal08- ENT-645 - Enterprise version bump

brittneyexline committed:
Merge pull request #17930 from edx/bexline/sso_id_verification2

ENT-945 Apply logic for taking into account SSOVerifications

yro committed:
Merge pull request #17953 from edx/yro/EDUCATOR-2644

Yro/EDUCATOR 2644

Commit:
Changes the mixin's delete algorithm.

Commit:
Allows UnregisteredLearnerCohortAssignments to delete_by_user_value.

Commit:
Merge pull request #18025 from edx/jmbowman/TE-2546

TE-2546 Remove obsolete requirements files

Commit:
Get MKTG_URL_LINK_MAP from ENV_TOKENS in CMS

waheedahmed committed:
Merge pull request #17985 from edx/waheed/LEARNER-4254-update-support-form

Update support form.

Commit:
Merge pull request #18034 from edx/roux/fix-deletable-by-user-value

Changes the DeletableByUserValue delete algorithm.

Commit:
Merge pull request #17415 from mahyard/master

Prepend unicode signature (BOM) to csv reports

Commit:
Merge pull request #18009 from edx/roux/unregisteredlearnercohortassignments-retire

UnregisteredLearnerCohortAssignments can retire users.

Commit:
Merge pull request #18011 from edx/sstudent/EDUCATOR-2731

make userorgtag deletable

MatthewPiatetsky committed:
Fix refundable status for audit orders

yro committed:
Merge pull request #17998 from edx/yro/EDUCATOR-2700

Yro/EDUCATOR 2700

AlasdairSwan committed:
Merge pull request #17945 from edx/alasdair/LEARNER-4880-gdpr-modal-display

LEARNER-4880 added confirmation modal for account deletion

jmbowman committed:
TE-2528 Upgrade testing dependencies

Commit:
Store Discussions ID Map on CourseDiscussionSettings model when course publish signal is fired.

saleem-latif committed:
Hide SSO related messages on login page if user is not in the SSO pipeline.

jaebradley committed:
upgrade cookie policy banner version

bmedx committed:
Merge pull request #17920 from edx/bmedx/fix_retirement_apis

Update retirement APIs to match spec

Commit:
Optimize dictionary combinations

Commit:
Merge pull request #18024 from edx/aj/LEARNER-4913_is_anon

Updates to use the is_anonymous flag of basket calculate

Commit:
Stop hitting mongo so much, cache this result

mulby committed:
add tests

Commit:
EDUCATOR-2618 request cache some more stuff

jaebradley committed:
EDUCATOR-2770: Update logout endpoint

jlajoie committed:
Merge pull request #18010 from edx/jlajoie/LEARNER-4700

LEARNER-4700: Converts credentials to plugin

brittneyexline committed:
Partial revert of "ENT-945 Update IDVerification interfaces to accountfor SSOVerification"

This partially reverts commit ee1c3a454851df8a06f751501d1f32c0cd9a6432.
The migration files introduced by the commit have been kept since they have been run
already on several enviornments.

michaelroytman committed:
update studio-frontend version to 1.8.0

Commit:
Merge pull request #18052 from edx/efischer/cache_stuff

Cache get_group_id_for_user

Mjloturco committed:
Merge pull request #18049 from edx/mloturco/learner-4715

retrap focus for seat unenroll survey a11y

learner-4715

Commit:
Merge pull request #18050 from edx/efischer/quick_fix

Optimize dictionary combination

Commit:
Merge pull request #18053 from edx/sstudent/cache_discussion_context

request cache some more stuff

iloveagent57 committed:
EDUCATOR-2771 | Adds an LMS API endpoint to retire a user account.

iloveagent57 committed:
EDUCATOR-2753 | Add calls to retire secondary user data to the /accounts/retire endpoint.

michaelroytman committed:
Merge pull request #18054 from edx/mroytman/bump-studio-frontend-1.8.0

update studio-frontend version to 1.8.0

feanil committed:
Remove this flaky test.

mulby committed:
fix pylint

feanil committed:
Merge pull request #18056 from edx/feanil/remove_flaky_test

Remove this flaky test.

adeelkhan committed:
Merge pull request #17973 from edx/adeel/learner_4344_catch_httperror_exception_thrown_psa

Catch HttpError thrown by Python Social Auth.

jaebradley committed:
Merge pull request #18059 from edx/cookie-policy-banner-1.1.6

Upgrade cookie policy banner version to 1.1.6

brittneyexline committed:
Merge pull request #18055 from edx/bexline/empty_sso_verification_migration

ENT-969: Partial revert of "ENT-945 Update IDVerification interfaces to account for SSOVerification"

Commit:
rate limit

Commit:
cache list of partitions in a course

Commit:
Merge pull request #18063 from edx/efischer/cache_partitions

Cache list of course partitions

edx-pipeline-bot committed:
Merge pull request #18062 from edx/release-mergeback-to-master

Merge release back to master

mulby committed:
Merge pull request #18028 from edx/mulby/support-cross-domain-ajax

support cross domain ajax requests to the experiments APIs

mikix committed:
Add some links to student records

Add some waffle-guarded connection points to the Credentials service
to start filling out the user flow for Student Records.

Specifically, add a button to the Program Progress Details page if
a certificate exists, and add a link in the Learner Profile page.

Both only appear if the 'student_records' waffle switch is active.

LEARNER-4701

jmbowman committed:
TE-2551 Fix commons.js optimization reliability

brittneyexline committed:
ENT-945 Using SSOVerifications in ID Verification flow

This reworks what was done #17930, since it had to be reverted from the IDVerificationAggregate migration.
We decided to abandon that model and directly read from both id verification models.

Commit:
Merge pull request #18067 from edx/jmbowman/TE-2551

TE-2551 Fix commons.js optimization reliability

Commit:
Merge pull request #18041 from edx/jmbowman/TE-2528

TE-2528 Upgrade testing dependencies

Commit:
Properly clean up a wrapped function

MatthewPiatetsky committed:
Merge pull request #18039 from edx/fix_refundable_status_for_audit

Fix refundable status for audit orders

coryleeio committed:
Merge pull request #18068 from edx/release-mergeback-to-master

Merge release back to master

nedbat committed:
Merge pull request #17368 from mitodl/pdpinch/remove-constansts-and-suffixes

Remove physical constants and suffixes from calc.py

nedbat committed:
Merge pull request #17748 from open-craft/jill/swift_content_length_from_fd_master

Set SWIFT_CONTENT_LENGTH_FROM_FD=False

stvstnfrd committed:
Remove unused VIRTUAL_UNIVERSITIES setting

It looks like this was part of the original rollout of the microsites
feature back in December 2013 [1], but was quickly removed in February
2014 [2]. These references have been orphaned ever since.
I can find no other references to it in the platform, save these.

[1] a3211a7405d92d09d59d3e2fe9ef437051324bcb
[2] aa43689a9398abdf1e28d6f8d509107be009b1c7

saleem-latif committed:
Merge pull request #18045 from edx/saleem-latif/ENT-908

ENT-908: Hide SSO related messages on login page if user is not in the SSO pipeline.

Rabia23 committed:
EDUCATOR-2774 add new condition for cert eligible column in grade report

nedbat committed:
Merge pull request #18069 from edx/nedbat/dont-bork-queryset-delete

Properly clean up a wrapped function

Commit:
Update models to 'Retire' not delete data

EDUCATOR-2700

feanil committed:
Remove these tests because they are flaky.

They all rely on modifying the handout which can sometimes fail.

nedbat committed:
Merge pull request #18071 from caesar2164/add-mktg-url-link-map-env-tokens

Get MKTG_URL_LINK_MAP from ENV_TOKENS in CMS

MatthewPiatetsky committed:
add has_staff_access to course metadata

michaelroytman committed:
pass courseDetails to editImageModal

waheedahmed committed:
Fix course selection on single support form.

Use defaultValue instead of value attribute.

LEARNER-5126

Commit:
Avoid fetching full group list repeatedly

stvstnfrd committed:
Remove unused META_UNIVERSITIES setting

This was removed back in July 2013 [1].

[1] e8ee29786fdfe2f39d780f0f64cc5592d19b0fb9

schenedx committed:
Merge pull request #18037 from edx/sofiya/ed-2706

Unlink CreditRequirementStatus PII for retired users

Commit:
Merge pull request #18061 from edx/sstudent/grading-policy-stuff

rate limit

feanil committed:
Merge pull request #18080 from edx/feanil/remove_flaky_handout_tests

Remove these tests because they are flaky.

feanil committed:
Merge pull request #18072 from stvstnfrd/clean/virtual-universities

Remove unused VIRTUAL_UNIVERSITIES setting

Commit:
Merge pull request #18084 from edx/efischer/optimize_again

Avoid fetching full group list repeatedly

coryleeio committed:
Merge pull request #18087 from edx/release-mergeback-to-master

Merge release back to master

Commit:
Process states param to gather all passed-in states.

Commit:
Update translations

bmedx committed:
Merge pull request #18014 from edx/bmedx/retirement_state_mgmt

Add a management command and settings to populate RetirementState models

schenedx committed:
Implement retire users function in the SurveyAnswer model for GDPR

MatthewPiatetsky committed:
Merge pull request #18081 from edx/exclude_instructors_from_paywall

add has_staff_access to course metadata

Commit:
emit ga events for user entitlement session actions

yro committed:
Merge pull request #18079 from edx/yro/EDUCATOR-2700-B

Update models to 'Retire' not delete data

Commit:
Add a test for multiple learner states.

schenedx committed:
Merge pull request #18091 from edx/schen/EDUCATOR-2698

Implement retire users function in the SurveyAnswer model for GDPR

Commit:
Merge pull request #18065 from edx/sofiya/ed2770

EDUCATOR-2770: Update logout endpoint to accept requests from user and validate password

Commit:
Merge pull request #18090 from edx/jeskew/fix_retirement_queue_method

Process states param to gather all passed-in states.

jaebradley committed:
LEARNER-4717 Have delete account modal button make request to deactivation endpoint

Rabia23 committed:
Merge pull request #18074 from edx/ri/EDUCATOR-2774-certificate-eligible-column

EDUCATOR-2774 add new condition for cert eligible column in grade report

brittneyexline committed:
Merge pull request #18070 from edx/bexline/sso_id_verification2

ENT-945 Using SSOVerifications in ID Verification flow

edx-cache-uploader-bot committed:
Updating db cache

estute committed:
Merge pull request #18094 from edx/testeng/bokchoy_auto_cache_update_f18a9d4e742f5e646d6176a845c96988e0240754

Bokchoy db cache update

Commit:
Update rate limit for enrollment api

LEARNER-5101: The current rate limit was adjusted to be
very high for staff users to account for excessive calls
from the ecommerce service. The ecommerce service has
been updated to not make these calls. This commit intends
to adjust the rate limit to it's original state before the
excessive calls.

Commit:
make password history retirable

mikix committed:
Fix extracting i18n from jsx code

React code (in jsx files) was not being extracted at all. Fixed this
by updating to edx-i18n-tools 0.4.5.

Also, template literals (feature of es2015) don't work well with
gettext at all. So convert those instances to HtmlUtils or
StringUtils interpolation calls.

FEDX-505

Commit:
This isn't used when rendering inline

schenedx committed:
Create deletion method for the PendingNameChange model to comply with GDPR

Commit:
remove tests also

schenedx committed:
Merge pull request #18100 from edx/schen/EDUCATOR-2695

Create deletion method for the PendingNameChange model to comply with GDPR

Commit:
Bump studio-frontend to v1.9.4 and fix bokchoy

Commit:
Merge pull request #18097 from edx/sstudent/EDUCATOR-2690

make password history retirable

Commit:
Merge pull request #18099 from edx/efischer/everything_is_dumb_always

Stop doing pointless things

Commit:
EDUCATOR-2690 bugfix

Commit:
Merge pull request #18020 from edx/ormsbee/mock_publish_mgt_cmd

Add simulate_publish management command.

Commit:
Remove activation keys upon logout of retired user.

robrap committed:
Merge pull request #18096 from edx/clee/ratelimit

Reduce rate limit for enrollment api

jaebradley committed:
Merge pull request #18101 from edx/LEARNER-4717-hook-delete-button-to-logout-api

LEARNER 4717: Call Retirement API from Delete Button

Commit:
Silence deprecation warnings

https://docs.djangoproject.com/en/1.11/releases/1.10/#using-user-is-authenticated-and-user-is-anonymous-as-methods

Commit:
Upgrade Django to 1.11.13

Commit:
Merge pull request #18103 from edx/sstudent/EDUCATOR-2690

EDUCATOR-2690 bugfix

Commit:
fix is_authenticated mocks

Commit:
Merge pull request #18107 from edx/jeskew/django_upgrade_1_11_13

Upgrade Django to 1.11.13

Commit:
EDUCATOR-2689: retire ManualEnrollmentAudit

Commit:
Adds "retire_user" to CreditRequest.

Commit:
Merge pull request #18106 from edx/efischer/is_it_this_easy

Silence deprecation warnings

arizzitano committed:
add base factory into textbooks factory

Commit:
Merge pull request #18111 from edx/sstudent/EDUCATOR-2689

EDUCATOR-2689: retire ManualEnrollmentAudit

arizzitano committed:
Merge pull request #18115 from edx/ari/cms-textbook-nav

add base factory into textbooks factory

Commit:
Merge pull request #18012 from edx/roux/oauth2-retire

EDUCATOR-2703: adds user retirement to OAuth2 tables

Commit:
cache results of thread authorship check

Commit:
Merge pull request #18102 from edx/fsheets/sfe-bump

Bump studio-frontend to v1.9.4 and fix bokchoy

Commit:
Add VNC info to testing docs

McKenzieW committed:
Merge pull request #18120 from edx/McKenzieW/learner-3830

emit ga events for user entitlement session actions

estute committed:
Merge pull request #18023 from edx/estute/TE-2427-reshard-lms4

Reshard python unittests

Commit:
Merge pull request #18114 from edx/roux/creditrequest_model_retirement

EDUCATOR-2658: Adds retire_user to CreditRequest.

Commit:
Prefetch user group data for bulk access checks

Commit:
Merge pull request #18122 from edx/fsheets/test-doc-vnc

Add VNC info to platform testing docs

nedbat committed:
Merge pull request #17810 from open-craft/afzaledx/move_site_verification_id_to_settings

Moved the ID used in the google-site-verification meta tag to settings

Commit:
swap page names for entitlement session analytics

michaelroytman committed:
Merge pull request #18085 from edx/mroytman/pass-course-details-to-image-modal

pass courseDetails to editImageModal

Commit:
Merge pull request #18104 from edx/jeskew/PLAT_2093_remove_activation_keys

Remove activation keys upon logout of retired user.

Commit:
Merge pull request #18121 from edx/efischer/one_more_time

Cache results of thread authorship check

nasthagiri committed:
Merge pull request #17602 from Microsoft/edx_master

Bulk Grades API (v1 grades api w/courses endpoint)

McKenzieW committed:
Merge pull request #18127 from edx/McKenzieW/learner-3830

swap page names for entitlement session analytics

Commit:
Fix heading structure

Commit:
Merge pull request #18125 from edx/efischer/the_code_also_gazes_into_you

Optimize _has_group_access for bulk checks

Commit:
Merge pull request #18128 from edx/douglashall/update_course_discussion_settings

Store Discussions ID Map on CourseDiscussionSettings model when course publish signal is fired.

Commit:
Merge pull request #18131 from edx/fsheets/test-doc-vnc

Fix heading structure

Commit:
Fix that when usage key none is not raise error.

Commit:
Merge pull request #18133 from edx/ai/EDUCATOR-2726_usage_key_none

Fix that when usage key none is not raise error.

bderusha committed:
Merge pull request #18000 from edx/password-rollout

Infrastructure for password policy compliance rollout

iloveagent57 committed:
EDUCATOR-2516 | Bump xblock-lti-consumer to 1.1.8

pdpinch committed:
expose attempts to python custom grader
